### DISCUSSION TITLE
Energy Added in HA Energy Dashboard

### DISCUSSION THREAD

**Discussion by siempie76** [alandtse/tesla#383] [Q&A]
Hi is there a way to get the energy added sensor in the Energy dashboard so I can get the associated cost when I charged the car at home?
It gives me fault

![image](https://user-images.githubusercontent.com/7797120/204750309-c424ad1b-9185-4785-a483-0fa64f556586.png)

**Comment by carleeno** [comment]
Gas consumption will not show electricity devices, if you try using "Individual devices" you'll see the right sensor. You're looking for `sensor.<car>_energy_added` (without the `_sensor` at the end).

I'm not sure why the old v2 sensor is still in the list, but you can ignore it. It seems to be a HA bug where the energy config shows deleted entities.

**Reply by siempie76** [reply]
it works in the "individual devices", but then I can't track the costs. A workaround is to put it in the gas consumption list so you can put a price entity with and don't mess up your true energy consumption & production. But somehow it doesn't show up there.

**Reply by carleeno** [reply]
Interesting, I noticed this link when trying to add a gas sensor: https://www.home-assistant.io/more-info/statistics/ which links further to long term statistics and the different configurations to use for energy sensors.

Looking at my logs, I see a warning for my car's energy sensor:
```
Entity sensor.mystique_energy_added from integration tesla_custom has state class total_increasing, but its state is not strictly increasing. Triggered by state 39.81 (39.83) with last_updated set to 2022-11-28T21:19:31.541802+00:00. Please report it to the custom integration author.
```

I suspect this is causing it to not be valid for long-term statistics. Can you see if you have same warning, or any other warnings regarding your energy_added sensor?

**Reply by siempie76** [reply]
yes I have the same. Statistics are not enabled or something.

I found this:
Hi! You can add any energy sensor there. If your energy sensor in available there to add, probably there is state_class: total_increasing missing on the entity.
Add this to customize.yaml:

sensor.your_entity_here:
  state_class: total_increasing
  friendly_name: FRIENDLY NAME
  device_class: energy

here: https://community.home-assistant.io/t/energy-cost-for-the-individual-devices/334932/59

**Reply by carleeno** [reply]
Sorry for the late reply.
Looking at the integration code, we already set the `state_class` to `total_increasing` and `device_class` to `energy`.

I played around with the attributes and discovered that in order to get the entity to be available for gas consumption, I had to change `device_class` to `gas`, however that prevents it from being added elsewhere. Additionally removing the `device_class` attribute causes an "Unexpected device class" error for the entity as well, so unfortunately we have to leave it set to `energy`.

I think for your workaround to add it as gas, you will need to create a template sensor which reads from the energy added sensor, then you can set your own `device_class` to `gas` for the template sensor

**Reply by siempie76** [reply]
I updated to HA 12.0 and now it is working!! I can add the Energy Added to the gas consumption. Thanks for your reply's
